src/engine/QueryPlanner.cpp

```diff
@@ -1758,6 +1776,14 @@ std::vector<QueryPlanner::SubtreePlan> QueryPlanner::createJoinCandidates(
     candidates.push_back(std::move(opt.value()));
   }
 
+  // if one of the inputs is the spatial join and the other input is a matching
+  // geometry, add the geometry as a child to the spatial join instead of
+  // creating a normal join
+  if (auto opt = createSpatialJoin(a, b, jcs)) {
+    candidates.push_back(std::move(opt.value()));
```

Comment:
The comment above is not super needed, but you need a comment as to why there is a `return`.
" Unbound spatial join operations are not compatible with "normal" joins, so we immediately return."